acceptance/tests/agent/agent_can_reconnect_after_master_failure.rb

```diff
@@ -0,0 +1,49 @@
+test_name 'C97323: agents can reconnect after master crash/restart' do
+
+  logfiles = {}
+  agents.each do |agent|
+    logfiles[agent.hostname] = agent.tmpfile(File.basename(__FILE__, '.*'))
+  end
+
+  teardown do
+    step 'stop puppet agent' do
+      on(agents, puppet_resource("service puppet ensure=stopped"))
+    end
+  end
+
+  step 'start puppet master' do
+    with_puppet_running_on(master, {}) do
+      agents.each do |agent|
+        step 'start puppet agent at rapid runinterval' do
+          on(agent, puppet("agent --server #{master.hostname} --runinterval 10 " +
+                           "--logdest #{logfiles[agent.hostname]} " +
+                           "--user #{agent['user']}"))
```

Comment:
I don't think it will work for windows agents, because `puppet agent ...` won't daemonize, but will run in the foreground every runinterval seconds. So I'd expect the beaker to block indefinitely while the agent runs.

To run async, you could set the runinterval in settings (`puppet config set ..`), and then use "puppet resource service puppet ensure=running". There also used to be a beaker helper for running an agent async.

I don't know if specifying logdest as a setting will work though, as I think it may require registry modifications to get the service control manager to pass --logdest at agent startup. /cc @Iristyle @glennsarti